pkg/pop/msal_confidential.go

```diff
@@ -0,0 +1,23 @@
+package pop
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/kubelogin/pkg/internal/pop"
+	"github.com/AzureAD/microsoft-authentication-library-for-go/apps/confidential"
+)
+
+func AcquirePoPTokenConfidential(
+	context context.Context,
+	popClaims map[string]string,
+	scopes []string,
+	cred confidential.Credential,
+	authority,
+	clientID,
+	tenantID string,
+	options *azcore.ClientOptions,
+	popKey *SwKey,
```

Comment:
Based on my previous comment, this might need a to be function instead of a struct. This way on your side you can set your timer and other caller (like kubelogin) can decide what logic they want to do such as having an unique key all the time